url/percent-encoding.window.js

```diff
@@ -0,0 +1,26 @@
+promise_test(() => fetch("resources/percent-encoding.json").then(res => res.json()).then(runTests), "Loading data…");
+
+function runTests(testUnits) {
+  for (const testUnit of testUnits) {
+    // Ignore comments
+    if (typeof testUnit === "string") {
+      continue;
+    }
+    for (const encoding of Object.keys(testUnit.output)) {
+      // Test UTF-8 in a special way
```

Comment:
It would be cleaner to just treat utf-8 the same as other encodings. Then, you wouldn't need to deal with both search and hash. Is there a reason you chose this architecture?

Reply:
I was thinking it would be good to always have the UTF-8 output, but not necessarily everything else. And to be able to assert that fragment (and perhaps path) parts of a URL are not encoded with this different encoding.

Reply:
Hmm. Keeping things simple is appealing to me, but I understand wanting more coverage. If you add sufficiently-long comments explaining the strategy and reasoning then I'd be happy.